### PR TITLE
Feasibility/xterm.js in webview

### PR DESCRIPTION
**Pull request by Riphiphip**
Moves serial monitor to a webview and incorporates [pc-xterm-lib](https://github.com/NordicPlayground/pc-xterm-lib) to enable code-sharing between different terminal apps.

### CODE REVIEW

**Review by bencefr — COMMENTED** [review]
All in all it looks good! Please make sure the code is linted, I see many questionable details that I'd expect a linter to catch and fix.

File: src/api.ts [hunk 1]
```diff
@@ -1,5 +1 @@
-import { SerialTerminal } from './serialTerminal';
-
-export const api = {
-    SerialTerminal,
-};
+export const api = {};
```
[the remark this answers is not in the document]
Comment (Riphiphip) [reply src/api.ts:1]: Will do

**Comment by bencefr** [comment]
The stream transform classes are very simple to cover by unit tests, please consider introducing such for those either here or in followup PR.

**Review by bencefr — APPROVED** [review]
Looks good to me.

**Review by deerob4 — APPROVED** [review]

**Pull request merged**